extensions/email/src/main/java/org/mule/extension/email/internal/operations/SetFlagOperation.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.email.internal.operations;
+
+import static java.lang.String.format;
+import static org.mule.extension.email.internal.util.EmailUtils.getAttributesFromMessage;
+import org.mule.extension.email.internal.EmailAttributes;
+import org.mule.extension.email.internal.exception.EmailException;
+import org.mule.runtime.api.message.MuleMessage;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.mail.Flags.Flag;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+
+/**
+ * Represents the set flag operation.
+ *
+ * @since 4.0
+ */
+public class SetFlagOperation
+{
+
+    /**
+     * Sets the specified {@code flag} into an email or {@link List} of emails
+     * that are being carried in the incoming {@link MuleMessage}
+     * <p>
+     * if no email(s) are found in the incoming {@link MuleMessage} the operation
+     * will fetch the email for the specified {@code emailNumber}.
+     * <p>
+     * If no email(s) are found in the {@link MuleMessage} and no {@code emailNumber} is specified.
+     * the operation will fail.
+     *
+     * @param muleMessage the incoming {@link MuleMessage}.
+     * @param folder      the configured folder.
+     * @param emailNumber the optional number of the email to be marked. for default the email is taken from the incoming {@link MuleMessage}.
+     * @param flag        the flag to be setted.
+     * @param expunge     if true, all mails marked as deleted will be erased from the folder.
```

Comment:
in which points of your code you have to open and close a folder? Is that something the connection provider should do when the connection is returned to the pool?

An operation to do that is fine, but maybe it makes sense to have it a config option so that the user doesn't have to call that operation every time